tests/bin/test_vyper.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+
+from vyper import (
+    __version__,
+)
+
+
+@pytest.fixture
+def make_tmp_vy_file(tmp_path):
+    name_counter = 0
```

Comment:
Using an `itertools.count` for this type of thing can be a bit cleaner in cases like this where you just need blind increments.  I *think* it allows you to remove the need for `nonlocal` and in theory you can inline the incrementing into the string formating via `f'{tmp{next(name_counter)}.vy'` if you're feeling brave.